### PR TITLE
Use version without build number for --agent-version;

### PR DESCRIPTION
**Pull request by ycliuhw**
Use version without build number for --agent-version for CI tests;

### CODE REVIEW

File: acceptancetests/jujupy/client.py [hunk 1]
```diff
@@ -583,6 +584,12 @@ def get_version_string_parts(version_string):
             return version_parts
 
 
+def strip_build_number(version_str):
+    # strip build number of the version string.
+    # for example, 2.9.27.888 -> 2.9.27
+    return '.'.join(version_str.split('.')[:3])
```
Comment (hpidcock) [inline acceptancetests/jujupy/client.py:590]: does this work for MAJ.MIN only? like `2.9` or `2.9-beta1`

**Review by wallyworld — APPROVED** [review]

File: acceptancetests/jujupy/client.py (hunk 1 above)
Comment (wallyworld) [reply acceptancetests/jujupy/client.py:590]: works for everything

Comment (wallyworld) [reply acceptancetests/jujupy/client.py:590]: actually, strip_build_number should probably got before the split('-')

Comment (ycliuhw) [reply acceptancetests/jujupy/client.py]: right, we have to cover the case of edge build with the OfficialBuild number like 2.9-beta1.888-ubuntu-amd64

**Pull request merged**